src/notation/view/notationviewinputcontroller.cpp

```diff
@@ -246,6 +252,79 @@ void NotationViewInputController::setViewMode(const ViewMode& viewMode)
     notation->setViewMode(viewMode);
 }
 
+constexpr qreal scrollStep = .8;
+constexpr qreal thinPadding = 10.0;
+constexpr qreal thickPadding = 25.0;
+
+void NotationViewInputController::moveScreen(int direction)
+{
+    auto notation = currentNotation();
+    if (!notation) {
+        return;
+    }
+    auto scale = m_view->currentScaling();
+    if (notation->viewMode() == ViewMode::LINE) {
+        m_view->moveCanvasHorizontal(m_view->width() * direction * scrollStep / scale);
+    } else {
+        m_view->moveCanvasVertical(m_view->height() * direction * scrollStep / scale);
+    }
```

Comment:
This implementation is actually quite a lot simpler than the one in MU3 (in ScoreView.cpp) - it has extra logic for frames that I couldn't quite figure out the need for.